### PR TITLE
Deprecate `wpo_wcpdf_due_date_display` hook

### LINKED ISSUE #883: Deprecate `wpo_wcpdf_due_date_display` hook
The `wpo_wcpdf_due_date_display` should be deprecated since it is used in a function for legacy templates.

### PR DESCRIPTION
**Pull request by MohamadNateqi**
close #883

### CODE REVIEW

File: includes/Main.php [hunk 1]
```diff
@@ -1646,8 +1646,19 @@ function_exists( 'WPO_WCPDF_Templates' ) &&
...
-		$due_date_title = is_callable( array( $document, 'get_due_date_title' ) ) ? $document->get_due_date_title() : __( 'Due Date:', 'woocommerce-pdf-invoices-packing-slips' );
+		$due_date = apply_filters_deprecated(
+			'wpo_wcpdf_due_date_display',
+			array(
+				date_i18n( wcpdf_date_format( $this, 'due_date' ), $due_date_timestamp ),
+				$due_date_timestamp,
+				$document_type,
+				$document
+			),
+			'3.8.7',
```
Comment (alexmigf) [inline includes/Main.php:1657]: Must be `3.9.0`

**Review by alexmigf — CHANGES_REQUESTED** [review]

File: includes/Main.php (hunk 1 above)
Comment (MohamadNateqi) [reply includes/Main.php]: Since this new function was introduced in version `3.8.7`, shouldn't we use this version here?

Comment (alexmigf) [reply includes/Main.php]: However, the old hook is still present, and there is no indication that it was deprecated in version `3.8.7`.

File: includes/Main.php [hunk 2]
```diff
@@ -1646,8 +1646,19 @@ function_exists( 'WPO_WCPDF_Templates' ) &&
 			return;
 		}
 
-		$due_date       = apply_filters( 'wpo_wcpdf_due_date_display', date_i18n( wcpdf_date_format( $this, 'due_date' ), $due_date_timestamp ), $due_date_timestamp, $document_type, $document );
-		$due_date_title = is_callable( array( $document, 'get_due_date_title' ) ) ? $document->get_due_date_title() : __( 'Due Date:', 'woocommerce-pdf-invoices-packing-slips' );
+		$due_date = apply_filters_deprecated(
```
[the remark this answers is not in the document]
Comment (MohamadNateqi) [reply includes/Main.php:1649]: @alexmigf If you're on the main branch of Premium Templates, then this function will return early, and the code should not reach this hook.
I'm confused how it reached this line on your end!

[the remark this answers is not in the document]
Comment (alexmigf) [reply includes/Main.php:1649]: I forgot to say that Pro branch was [this](https://github.com/wpovernight/woocommerce-pdf-ips-pro/pull/459).

[the remark this answers is not in the document]
Comment (MohamadNateqi) [reply includes/Main.php:1649]: Still, it should not reach that line!
That PR in the Pro is just using that filter. On the main branch of the Premium Templates extension, and this PR branch in the free plugin, this [line](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/884/files/3e34e553d9cb6a1a412cf74a1d0f18e47f019b90#diff-bd49c6b6fbe7a6920acaf94a4a3de786f05bd21c2a67f9cef8d8b1f40d54971bR1634) should return early and prevent applying this deprecated filter.

Could you please explain to me how this occurs on your end?

[the remark this answers is not in the document]
Comment (alexmigf) [reply includes/Main.php:1649]: Weird, now seems to be working, not sure what happened.

**Review by alexmigf — APPROVED** [review]
:rocket:

**Pull request merged**